linera-base/src/identifiers.rs

```diff
@@ -169,22 +169,15 @@ pub enum BlobType {
 
 impl Display for BlobType {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            BlobType::Data => write!(f, "Data")?,
-        };
-
-        Ok(())
+        write!(f, "{}", serde_json::to_string(self).unwrap())
```

Comment:
I think there's a dedicated `to_writer` function for this. Also, let's avoid `unwrap`ping:

```suggestion
        serde_json::to_writer(f, self).map_err(|_| fmt::Error)
```

Reply:
Afaik, the `to_writer()` needs the `f` implement the `std::io::Write` trait. 
But, `std::fmt::Formatter` does not meet the requirements.
I believe we should stick to `to_string()`. Then, maybe, we cannot avoid `unwrap`.

Reply:
But `Formatter` _does_ imply `Write`, doesn't it?
https://doc.rust-lang.org/nightly/std/fmt/struct.Formatter.html#impl-Write-for-Formatter%3C'_%3E

And we can always avoid `unwrap` using `.map_err(|_| fmt::Error)`.

Reply:
Is that `Write` equal to `std::fmt::Write`? not the `std::io::Write`.
`serde_json::to_writer` needs the writer to implement the `std::io::Write`, targeting the I/O stream.
https://docs.rs/serde_json/latest/serde_json/fn.to_writer.html

Also, if we really wanna avoid `unwrap`, how about this?
```
impl Display for BlobType {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match serde_json::to_string(self) {
            Ok(s) => write!(f, "{}", s),
            Err(_) => return Err(fmt::Error),
        }
    }
}
```

Reply:
Ah, I see! Right, I guess we can't use `to_writer` then.

Sure, I'm fine with the `match`, too, if you prefer that. :+1: